code/segmentedtickerplant/stplog.q

```diff
@@ -201,6 +201,7 @@ endofperiod:{[currentpd;nextpd;data]
   .lg.o[`endofperiod;"executing end of period for ",.Q.s1 `currentperiod`nextperiod!(currentpd;nextpd)];
   .stpps.endp[currentpd;nextpd;data];                   // sends endofperiod message to subscribers
   currperiod::nextpd;                                   // increments current period
+  nextperiod::multilogperiod+currperiod;                // increments next period
```

Comment:
`nextperiod` is already updated inside `stpeoperiod` (line 214) - are we sure this won't cause a double increment?

Reply:
As I understand, `endofperiod[]` is intended to be called on chained tickerplants, immediately downstream from `stpeoperiod[]` on the "core" TP.
Consequently, `stpeoperiod[]` and `endofperiod[]` update the `nextperiod` values on their _specific_ processes; alternatively, `nextperiod` on a given process is only updated by one of `stpeoperiod[]` and `endofperiod[]`.

Reply:
From what I can determine, `nextperiod` is not utilised by the chained TP. However, `currperiod` is, as a consequence of the call to `periodrollover[]` on line 205; I feel that keeping `nextperiod` updated to a "sensible" counterpart value is safer & more future-proof than leaving it unchanged from its initial definition as part of running `init[]`.

The alternative is adding logic to `init[]` to only define `nextperiod` if its called on a chained TP process, but that may be another ticket/PR entirely.

Reply:
If `nextperiod` isn't used by chainedtp (which IMO, it shouldn't be - CTP does not control it's own rollover, so it never needs to know when the next period starts), I don't think we should be adding code to update it for no reason.

Reply:
Okay, I can revert this addition then - I feel a further update to not have the chainedtp define nextperiod at all may help conserve a sliver of memory, but that's for another time.